core/src/main/java/org/elasticsearch/repositories/Repository.java

```diff
@@ -72,6 +72,15 @@
     /**
      * Reads snapshot description from repository.
      *
+     * @param repositoryData repository data
+     * @param snapshotId  snapshot id
+     * @return information about snapshot
+     */
+    SnapshotInfo getSnapshotInfo(RepositoryData repositoryData, SnapshotId snapshotId);
```

Comment:
I appreciate that requiring to pass RepositoryData on getSnapshotInfo might not be the cleanest solution regarding Repository's API.